backend/repositories/embeding_model.py

```diff
@@ -43,29 +50,38 @@ def create_embedding_model(self, embedding_model: EmbeddingModel) -> APIError |
             self.db_session.commit()
             return None
         except Exception as e:
+            logger.error(f"Error creating embedding model: {e}")
             self.db_session.rollback()
-            return APIError(detail=str(e))
-            
-    def update_embedding_model(self, id: int, embedding_model: EmbeddingModel) -> APIError | None:
+            return APIError(err_code=20001)
+
+    def update_embedding_model(
+        self, id: int, embedding_model: EmbeddingModel
+    ) -> APIError | None:
         """
         Update embedding model
         """
         try:
-            self.db_session.query(EmbeddingModel).filter(EmbeddingModel.id == id).update(embedding_model.model_dump())
+            self.db_session.query(EmbeddingModel).filter(
+                EmbeddingModel.id == id
+            ).update(embedding_model.model_dump())
             self.db_session.commit()
             return None
         except Exception as e:
+            logger.error(f"Error updating embedding model: {e}")
             self.db_session.rollback()
-            return APIError(detail=str(e))
-    
+            return APIError(err_code=20001)
```

Comment:
What is 20001 err_code, change to an ENUM

Reply:
But I will create a PR for this error handler feature.